### PR TITLE
feat(pallet-gear): Added example and test to test signal weak case

### LINKED ISSUE #2814: Check signal weak case
### Problem to Solve

Write test to check behaviour if signal calls `system_reserve() && wait` and end up in waitlist

### Possible Solution

Write test and check

### Notes

_No response_

### PR DESCRIPTION
**Pull request by mqxf**
Resolves #2814 

@reviewer-or-team

### CODE REVIEW

File: examples/signal-wait/src/wasm.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+static mut FIRST_EXEC: bool = true;
+
+#[no_mangle]
+extern "C" fn handle() {
+    if unsafe { FIRST_EXEC } {
+        unsafe {
+            FIRST_EXEC = false;
+        }
+        gstd::exec::system_reserve_gas(1_000_000_000).unwrap();
```
Comment (StackOverflowExcept1on) [inline examples/signal-wait/src/wasm.rs:29]:
```suggestion
        exec::system_reserve_gas(1_000_000_000).unwrap();
```

**Review by SabaunT — COMMENTED** [review]
lgtm, but I'd re-write demo using `demo-constructor`. @breathx could you please check the test itself?

File: pallets/gear/src/tests.rs [hunk 2]
```diff
@@ -14660,6 +14660,52 @@ fn test_gas_info_of_terminated_program() {
...
+
+        run_to_next_block(None);
+
+        assert_ok!(GasHandlerOf::<Test>::get_system_reserve(mid));
+        assert!(WaitlistOf::<Test>::contains(&pid, &mid));
+
+        run_to_next_block(None);
+
+        let signal_mid = MessageId::generate_signal(mid);
+        assert!(WaitlistOf::<Test>::contains(&pid, &signal_mid));
```
Comment (breathx) [inline pallets/gear/src/tests.rs:14705]: so whats the aim of the test?
its in checking that everything works well when signal_mid ends up in waitlist as outdated, so you need prolong test

**Review by breathx — CHANGES_REQUESTED** [review]

**Comment by breathx** [comment]
> lgtm, but I re-write demo using `demo-constructor`. @breathx could you please check the test itself?

yep, demo can be rewritten with demo_constructor

**Comment by mqxf** [comment]
The test should be correct, going to rewrite using demo-constructor

**Review by breathx — APPROVED** [review]

**Review by breathx — CHANGES_REQUESTED** [review]
Whats happening with the diff?

**Comment by mqxf** [comment]
The diff was breaking because of changes in master I assume.  I just merged master into this and its back to normal

File: pallets/gear/src/tests.rs [hunk 3]
```diff
@@ -14697,6 +14697,126 @@ fn test_gas_info_of_terminated_program() {
     })
 }
 
+#[test]
+fn test_handle_signal_wait_old() {
```
Comment (breathx) [inline pallets/gear/src/tests.rs:14701]: merge initial version with this pr, please
in several - adapt it to constructor

**Review by breathx — CHANGES_REQUESTED** [review]

**Review by breathx — APPROVED** [review]

**Comment by breathx** [comment]
@SabaunT ptal

**Review by SabaunT — APPROVED** [review]

**Pull request merged**